app/src/main/java/app/f3d/F3D/android/Utils/FileInteractionContract.java

```diff
@@ -0,0 +1,29 @@
+package app.f3d.F3D.android.Utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.activity.result.contract.ActivityResultContract;
+
+public class FileInteractionContract extends ActivityResultContract<Void, Uri> {
+
+    @Override
+    public Intent createIntent(Context context, Void input) {
+        // Create an intent to open files with supported mime types
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.setType("*/*");
```

Comment:
Does this line mean we are accepting all mimetypes? If so, why is the line below is needed?

Reply:
No We dont accept all type,

initally we set as all types, but we restrict it under Intent.putExtra parameter.

Without using setType the app will crash as it doesn't know in the begining what is allowed

Reply:
got it!